third-party/spamassassin/src/main/java/org/apache/james/spamassassin/SpamAssassinInvoker.java

```diff
@@ -239,8 +241,11 @@ private boolean isSpam(String line) {
             .flatMap(connection -> {
                 LOGGER.debug("Report mail as {}", messageClass);
                 return Mono.fromCallable(() -> IOUtils.toByteArray(message)).flatMap(
```

Comment:
while you are at it, you should get rid of that byte[] reading
I had a look and the root of the problem is that `SpamAssassinListener` is not passing the message size despite having the information at hand.
Could you fix that also? There are speed and memory usage gains to expect